module/VuFind/tests/unit-tests/src/VuFindTest/Navigation/AccountMenuTest.php

```diff
@@ -0,0 +1,251 @@
+<?php
+
+/**
+ * Account menu tests.
+ *
+ * PHP version 8
+ *
+ * Copyright (C) The National Library of Finland 2024.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Aleksi Peebles <aleksi.peebles@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public
+ *           License
+ * @link     https://vufind.org/wiki/development:testing:unit_tests Wiki
+ */
+
+namespace VuFindTest\Navigation;
+
+use VuFind\Auth\ILSAuthenticator;
+use VuFind\Auth\Manager;
+use VuFind\Config\AccountCapabilities;
+use VuFind\ILS\Connection;
+use VuFind\Navigation\AccountMenu;
+
+use function count;
+
+/**
+ * Account menu tests.
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Aleksi Peebles <aleksi.peebles@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public
+ *           License
+ * @link     https://vufind.org/wiki/development:testing:unit_tests Wiki
+ */
+class AccountMenuTest extends \PHPUnit\Framework\TestCase
+{
+    /**
+     * Test that the menu is the default menu if configuration is missing.
+     *
+     * @return void
+     */
+    public function testMissingConfiguration()
+    {
+        $this->assertEquals(
+            $this->getAccountMenu()->getMenu(),
+            $this->getAccountMenu(AccountMenu::getDefaultMenuConfig())->getMenu()
+        );
+    }
+
+    /**
+     * Test the default menu when all check methods return false.
+     *
+     * @return void
+     */
+    public function testDefaultMenuAllCheckMethodsReturnFalse()
+    {
+        $menu = $this->getAccountMenu(
+            AccountMenu::getDefaultMenuConfig(),
+            $this->getCheckMethods(false)
+        )->getMenu();
+        $this->assertTrue(
+            count($menu['Account']['MenuItems']) === 1
+            && reset($menu['Account']['MenuItems'])['label'] === 'Profile'
+        );
+    }
+
+    /**
+     * Test backward compatibility for old configurations.
+     *
+     * @return void
+     */
+    public function testBackwardCompatibilityForOldConfigurations()
+    {
+        $menu = $this->getAccountMenu($this->getOldDefaultMenuConfig())->getMenu();
+        $this->assertTrue(count($menu['Account']['MenuItems']) === 12);
```

Comment:
You can use assertCount here:

```suggestion
        $this->assertCount(12, $menu['Account']['MenuItems']);
```